### PR TITLE
Hydra updates

### PR DESCRIPTION
**Pull request by lheckemann**
This PR combines changes from #157072, #152427, and a further update to a hydra commit from last week.

TODOs:
- [x] Decide whether to keep the refactoring, or continue to provide for multiple versions of hydra. I'm leaning towards keeping the refactor in spite of the code churn it causes. The prospect of another version of hydra being reintroduced seems unlikely.
- [ ] Remove multi-versioning from the hydra NixOS tests
- [x] Potentially clean up commits (some changes, such as the maintainership changes, should probably not be rolled into the commits they currently lie in)

cc @Mindavi @dasJ who have agreed to be co-maintainers, @lovesegfault who opened the refactoring PR.

Closes #157072, closes #152427.

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -17241,8 +17241,7 @@ with pkgs;
 
   hwloc = callPackage ../development/libraries/hwloc {};
 
-  inherit (callPackage ../development/tools/misc/hydra { })
-    hydra-unstable;
+  hydra_unstable = callPackage ../development/tools/misc/hydra/unstable.nix { nix = nix_2_4; };
```
[the remark this answers is not in the document]
Comment (lovesegfault) [reply pkgs/top-level/all-packages.nix:17244]: Is there a good reason to be using 2_4? I imagine this should just be the latest stable?

**Comment by Mindavi** [comment]
Gonna try and run this for a bit to see if it works as expected, ping me if you'd like to hear how it went :).

I'm in favor of removing the multi-versioning, so :+1: from me on that one. IMO the commit history is clean enough, and I think this could be acceptable wrt the policy (but I'd recommend to check it if you don't have it fresh in your mind).

File: pkgs/top-level/all-packages.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (lheckemann) [reply pkgs/top-level/all-packages.nix:17244]: This was before https://github.com/NixOS/hydra/pull/1140 was merged, if we update the package accordingly it should be fine to use at least 2.6 if not 2.7.

**Comment by Mindavi** [comment]
Finally got this building (was having some issues with something nvidia related, not sure why it started building fine all of a sudden).

Everything seems to be fine on first glance, started a rebuild of something and that seemed to work fine. No obvious issues otherwise. Will keep this running for a while and report back if I see anything problematic.

File: pkgs/top-level/all-packages.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (mweinelt) [reply pkgs/top-level/all-packages.nix:17244]: hydra.nixos.org shows garbage collection problems with 2.5 and 2.6, so staying on 2.4 might be the better choice.

**Comment by Mindavi** [comment]
There's a merge conflict that should be solved.

**Comment by Mindavi** [comment]
Started rebasing: https://github.com/Mindavi/nixpkgs/tree/hydra-updates-2.

Time's up for now, might look at this for a bit this evening.

**Pull request closed without merging**